analyses/malawi/dryspells_trigger/data-viz.R

```diff
@@ -0,0 +1,70 @@
+
+#####
+## setup
+#####
+
+# load libraries
+library(tidyverse)
+library(sf)
+library(raster)
+
+# set options
+rasterOptions(maxmemory = 1e+09)
+options(scipen = 999)
+
+# set directory paths
+# AA_DATA_DIR is set as a variable in .Renviron or .bashprofile
+data_dir <- Sys.getenv("AA_DATA_DIR")
+shapefile_path <- paste0(data_dir, "/public/raw/mwi/cod_ab/mwi_adm_nso_20181016_shp")
+plot_path <- paste0(data_dir, "/public/processed/mwi/plots/dry_spells/dry_spell_plots")
+
+# import dry spells groundtruth dataset
+dry_spells_groundtruth_dataset_url <- "https://data.humdata.org/dataset/df68065d-704c-4556-b111-394df000cee4/resource/951c0454-23ab-4057-a7fb-7cfeed4f842d/download/full_list_dry_spells.csv"
+ds <- read.csv(dry_spells_groundtruth_dataset_url)
+
+# read in country shapefiles
+mwi_adm2 <- st_read(paste0(shapefile_path, "/mwi_admbnda_adm2_nso_20181016.shp"))
+
+#####
+## map  historical dry spells
+#####
+
+# summarise dry spells frequency per district / year
+ds_counts_per_year <- ds %>%
+  filter(season_approx != "outside rainy season") %>%
+  group_by(pcode, season_approx) %>%
+  summarise(n_ds = n())
+
+ds_counts <- ds %>%
+  filter(season_approx != "outside rainy season") %>%
+  group_by(pcode) %>%
+  summarise(n_ds = n())
+
+# combine dry spells groundtruth dataset and shapefiles
+data <- mwi_adm2 %>%
+  left_join(ds_counts, by = c("ADM2_PCODE" = "pcode"))
+
+# create map
+
+data %>%
+  ggplot() +
+  geom_sf(aes(fill = n_ds)) +
+  scale_fill_continuous(type = "viridis", "Total number of events") +
```

Comment:
Minor thing, but wouldn't use `viridis` as a color scale but instead a sequential scale such that it is clear that a darker hue indicates more events.

Reply:
I think I had started with that and found the hues unclear. I'll try again to confirm.

Reply:
It actually looks fine; putting the blue hues in and reversed it so darker hues represent higher counts.